L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorSuperTriggerCellImpl.cc

```diff
@@ -36,62 +41,58 @@ HGCalConcentratorSuperTriggerCellImpl::getSuperTriggerCellId(int detid) const {
     }
     else{
       int TC_wafer = TC_idV8.wafer();
-      int TC_12th = ( TC_idV8.cell() & kSplit12_ );
-      int TC_3rd = ( TC_idV8.cell() & kSplit3_ );
-      
       int thickness = triggerTools_.thicknessIndex(detid,true);
-      int TC_split = TC_12th;
-      if (stcSize_.at(thickness) == kSTCsize16_) TC_split = TC_3rd;
-      
+      int TC_split = ( TC_idV8.cell() & kSplit_[ stcSize_.at(thickness) ] );
+
       return TC_wafer<<kWafer_offset_ | TC_split;
     }
 
   }
 
   else if ( TC_id.det() == DetId::HGCalTrigger ){//V9
     
-    HGCalTriggerDetId TC_idV9(detid);
-
     if( triggerTools_.isScintillator(detid) ){
-      return TC_idV9.triggerCellU() << kWafer_offset_ | TC_idV9.triggerCellV(); //scintillator
+      HGCScintillatorDetId TC_idV9(detid);
+      return TC_idV9.ieta() << kWafer_offset_ | TC_idV9.iphi(); //scintillator
```

Comment:
Careful, `ieta` can be negative. Better to use `ietaAbs`.
Also the `iphi` mask in `HGCScintillatorDetId` is `0x1ff`, so a 6 bit shift won't be enough. Better to use directly the offset defined in the detid.